RecoMuon/CosmicMuonProducer/src/CosmicMuonTrajectoryBuilder.cc

```diff
@@ -133,7 +133,7 @@ void CosmicMuonTrajectoryBuilder::setEvent(const edm::Event& event) {
 }
 
 MuonTrajectoryBuilder::TrajectoryContainer CosmicMuonTrajectoryBuilder::trajectories(const TrajectorySeed& seed) {
-  vector<Trajectory*> trajL = vector<Trajectory*>();
+  TrajectoryContainer emptyContainer;
```

Comment:
The compiler will not allow this to be `const TrajectoryContainer` because that would require a copy constructor for `std::unique_ptr`.

Reply:
The new implementation "intentionally" prevents return value optimization (the previous one allowed it).

Reply:
Playing around with compiler explorer seems to show minimal differences between reusing a container and creating a new one:

https://godbolt.org/z/k36w2x

The difference appears to be a second call to a default constructor. In all cases a move constructor is still called. I originally used a std::vector<int> as the container but with all the inlining, it was difficult to tell exactly what the compiler was doing (although it appeared to also just calling an additional default constructor call).